src/routes/sessions/[id]/+page.svelte

```diff
@@ -314,14 +341,27 @@
 						{/key}
 
 						<nav class="prompt-editor__toolbar">
+							{#if messageIndexToEdit !== null}
+								<Button
+									variant="outline"
+									on:click={() => {
+										prompt = '';
+										messageIndexToEdit = null;
+									}}
+								>
+									{$i18n.t('cancel')}
+								</Button>
+							{/if}
 							<ButtonSubmit
 								{handleSubmit}
 								hasMetaKey={isPromptFullscreen}
 								disabled={!prompt ||
 									!$settingsStore?.ollamaModels.length ||
 									!$settingsStore?.ollamaModel}
 							>
-								{$i18n.t('sessionsPage.run')}
+								{$i18n.t(
+									messageIndexToEdit !== null ? 'sessionsPage.saveAndRun' : 'sessionsPage.run'
+								)}
 							</ButtonSubmit>
```

Comment:
Here's how I think the `handleSubmit` logic should work.

I also flipped the ternary operators to avoid using a negative evaluation (`!== null`), which makes the code less brittle.

```suggestion
							<ButtonSubmit
								handleSubmit={messageIndexToEdit === null
									? handleSubmitNewMessage
									: handleSubmitEditMessage}
								hasMetaKey={isPromptFullscreen}
								disabled={!prompt ||
									!$settingsStore?.ollamaModels.length ||
									!$settingsStore?.ollamaModel}
							>
								{$i18n.t(
									messageIndexToEdit === null ? 'sessionsPage.run': 'saveAndRun'
								)}
							</ButtonSubmit>
```

Reply:
I understand, but I added this reactive function because the old `handleSubmit` was not only used at the `ButtonSubmit`, but also at `handleKeyDown` and `FieldTextEditor`, so we would have that logic triplicated.

Reply:
Oh, I see.. In that case can we keep `handleSubmit()` as it was and do the check for _Editing_ or _New_ inside of it once it gets called?

I'd like to avoid having a reactive function if possible.

Reply:
Sure! Is [this](https://github.com/fmaclen/hollama/pull/170/commits/e78d5477d157adaf10ee423b33f44aaf11addc84) what you mean?

Reply:
Yup, in general I try to avoid the reactive syntax unless we absolutely need to.

Also, not sure if you heard about the concept of ["Runes"](https://svelte-5-preview.vercel.app/docs/runes) coming up in Svelte 5... but I think that's going to require refactoring most (if not all) of our reactive declarations, I expect  that it's not going to be fun migrating from 4 to 5 😬